socs/agents/ld_monitor/agent.py

```diff
@@ -0,0 +1,424 @@
+
+                self.newdata_dict = {
+                    'd_type': 1,
+                    'time_last': time.time(),
+                    'dist': int(data_split[1]),
+                    'unit_d': unit_d
+                }
+                self.data_dict.update(self.newdata_dict)
+
+                return self.data_dict
+
+            # "high e-field" sentence, account for 2 types
+            elif param == 'HF':
+                data_split = self.data[1:].split(',')
+                if len(data_split) == 1:
+                    self.newdata_dict = {
+                        'd_type': 2,
+                        'high_field': 1,
+                        'hifield_value': float(self.data_dict['field_value'])
+                    }
+                else:
+                    self.newdata_dict = {
+                        'd_type': 2,
+                        'hifield_value': float(data_split[1])
+                    }
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+            # "status" sentence
+            elif param == 'ST':
+                faultcode = int(self.data.split(',')[-1].split('*')[0], 16)
+                data_split = [int(i) for i in self.data.split(',')[1:-1]]
+
+                self.newdata_dict = {
+                    'd_type': 3,
+                    'alarm_r': data_split[0],
+                    'alarm_o': data_split[1],
+                    'alarm_y': data_split[2],
+                    'delay_g': data_split[3],
+                    'clear': data_split[4],
+                    'r_timer': data_split[5],
+                    'o_timer': data_split[6],
+                    'y_timer': data_split[7],
+                    'g_timer': data_split[8],
+                    'allclear_timer': data_split[9],
+                    'faultcode': faultcode
+                }
+
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+            # disregard "alarm timers" sentence but update sentence type
+            elif param == 'WT':
+                self.newdata_dict = {'d_type': 4}
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+    def read_cycle(self):
+        """
+        In each cycle data is read and then parsed following
+        the format required to publish data to the ocs feed
+        """
+        try:
+            cycle_data = {}
+            self.read_data()
+
+            # updates time since last strike if previous strike data exists
+            if self.data_dict['time_last'] == -1.:
+                self.data_dict['tsince_last'] = -1.
+            else:
+                self.data_dict['tsince_last'] = (time.time()
+                                                 - self.data_dict['time_last'])
+
+            # parse data to ocs agent feed format
+            for key in self.data_dict:
+                cycle_data[key] = {'value': self.data_dict[key]}

```

Comment:
This seems to add an extra layer of dictionary to store the value. I would suggest simply returning the `self.data_dict` as is, since that then simplifies the loop in the `acq()` Process down on line 341 to something like:
```
for reg in regdata:
    data['data'][reg] = regdata[reg]
    field_dict = {reg: regdata[reg]}
    session.data['fields'].update(field_dict)
session.data.update({'timestamp': current_time})
```

Even better (feel free to use more descriptive names for "key" and "value"):
```
for key, value in regdata.items():
    data['data'][key] = value
    field_dict = {key: value}
    session.data['fields'].update(field_dict)
session.data.update({'timestamp': current_time})
```

Looking at this simplified version it becomes clear that `field_dict == regdata`, this further reduces to:
```
for key, value in regdata.items():
    data['data'][key] = value

session.data['fields'].update(regdata)
session.data.update({'timestamp': current_time})
```